Detect system-installed Rt* libraries

OPNA2608 opened the pull request:
Closes #185.

- RtAudio & RtMidi: Uses pkg-config in CMake to find these libraries & required settings, falls back to finding the libraries & header directories via CMake-native mechanisms (`find_library` & `find_path`)
- RtAudio: Checks if the found headers have the required function `RtAudio::getCompiledApiName` (introduced with 5.1.0), rejects the detected results if they don't
- JACK & PulseAudio: Fallback if pkg-config is not available
- JACK: Also detect on Darwin

Builds & runs on Ubuntu 18.04 with only RtAudio checked out (RtAudio is still version 5.0.0 there, so only system-RtMidi can be used). I'll test this on Darwin & a more updated Linux some time this week.

Do you want a CMake option to control this detection behaviour? If yes, what should the default value be?

Wohlstand commented:
The detection would be fine to be the default, and then have both options:
- fallback into a shipped build of the library
- the manually-set flag that will enforce use the shipped build of the library always

Wohlstand commented:
P.S. The conflict here is because I had to comment out the code that enables the ASIO support on Windows, that doesn't work properly, unfortunately, and people complained me several problems with it:
- just no audio
- startup crash
So, I desired to disable this at all. At the same time, this is a proprietary technology that is not met by the FOSS community.

OPNA2608 commented:
- Rebased to fix merge conflict
- Added `USE_VENDORED_RTMIDI` and `USE_VENDORED_RTAUDIO` CMake options to disable the library detection (default value is `OFF`)

Wohlstand commented:
Good! :fox_face: :+1:

Wohlstand commented on CMakeLists.txt:
```diff
@@ -271,68 +266,176 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      set(CMAKE_REQUIRED_INCLUDES ${RTAUDIO_MANUAL_INCLUDE_DIRS})
+      set(CMAKE_REQUIRED_LIBRARIES ${RTAUDIO_MANUAL_LIBRARY})
+      check_cxx_symbol_exists(RtAudio::getCompiledApiName "RtAudio.h" RTAUDIO_MANUAL_SUFFICIENT)
+      if(NOT RTAUDIO_MANUAL_LIBRARY-NOTFOUND AND NOT RTAUDIO_MANUAL_INCLUDE_DIRS STREQUAL RTAUDIO_MANUAL_INCLUDE_DIRS-NOTFOUND AND RTAUDIO_MANUAL_SUFFICIENT)
```
> `NOT RTAUDIO_MANUAL_LIBRARY-NOTFOUND`

Isn't a mistake? :thinking: 
Please put strings into quotes also as they confuse some things here.

OPNA2608 replied:
I think I misread the docs w/r/t `find_library`'s `<var>-NOTFOUND` :man_facepalming:, will fix.

Wohlstand replied:
Usually goes:
```cmake
find_library(KURWA kurwa)

if(KURWA)
     message("Found!")
endif()

if(NOT KURWA)
     message("Kurwa mać!")
endif()
```

OPNA2608 replied:
Was quite the hot day when I hacked that together, should be better now. :sweat_smile:

Wohlstand replied:
Now looks much better! :fox_face: :+1:

OPNA2608 commented:
Should I port this to OPN2BankEditor or will you do that?

Wohlstand commented:
If you want, feel free to try :fox_face: 
Also please apply my patch to disable the ASIO on the OPN2 side too, it still being enabled here.
